### DISCUSSION TITLE
auth guards in api

### DISCUSSION THREAD

**Discussion by ateeq1999** [adonisjs/core#2090] [Help]
{
  "message": "Cannot login user. Value of \"id\" is not defined",
  "stack": "Exception: Cannot login user. Value of \"id\" is not defined\n    at OATGuard.getUserForLogin (E:\\Projects\\api_adonis\\node_modules\\@adonisjs\\auth\\build\\src\\Guards\\Base\\index.js:104:19)\n    at OATGuard.login (E:\\Projects\\api_adonis\\node_modules\\@adonisjs\\auth\\build\\src\\Guards\\Oat\\index.js:238:35)\n    at OATGuard.attempt (E:\\Projects\\api_adonis\\node_modules\\@adonisjs\\auth\\build\\src\\Guards\\Oat\\index.js:208:21)\n    at CompanyAuthController.login (E:\\Projects\\api_adonis\\app\\Controllers\\Http\\CompanyAuthController.ts:11:19)\n    at Object.PreCompiler.runRouteHandler [as fn] (E:\\Projects\\api_adonis\\node_modules\\@adonisjs\\http-server\\build\\src\\Server\\PreCompiler\\index.js:46:31)\n    at Server.handle (E:\\Projects\\api_adonis\\node_modules\\@adonisjs\\http-server\\build\\src\\Server\\index.js:136:13)"
}

**Comment by crbast** [comment]
Can you please share your code (and `auth` configuration)? It's hard to help you without information, thx.

https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code
